Source/SwiftLintFramework/Rules/ForceUnwrappingRule.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  ForceUnwrappingRule.swift
+//  SwiftLint
+//
+//  Created by Benjamin Otto on 14/01/16.
+//  Copyright © 2016 Freeletics GmbH. All rights reserved.
+//
+
+import SourceKittenFramework
+
+public struct ForceUnwrappingRule: Rule, OptInRule {
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "force_unwrapping",
+        name: "Force Unwrapping",
+        description: "Force unwrapping should be avoided.",
+        nonTriggeringExamples: [
+            "if let url = NSURL(string: query)",
+            "navigationController?.pushViewController(viewController, animated: true)"
+        ],
+        triggeringExamples: [
+            "let url = NSURL(string: query)↓!",
+            "navigationController↓!.pushViewController(viewController, animated: true)",
+            "let unwrapped = optional↓!",
+            "return cell↓!"
+        ]
+    )
+
+    public func validateFile(file: File) -> [StyleViolation] {
+        // Get all identifiers from syntax map
+        let identifiers = file.syntaxMap.tokens.filter({
+                $0.type == SyntaxKind.Identifier.rawValue
+            }).flatMap({
+                file.contents.substringWithByteRange(start: $0.offset, length: $0.length)
+            })
+        // Check if there is any identifier followed by a '!'
+        let violations = Set(identifiers).flatMap({
```

Comment:
This is going to be terribly inefficient, since you're scanning the entire file for a regex match for each identifier in the file.